addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -1323,10 +1350,35 @@ MockServer.include({
      * @param {integer} message_id
      */
     _mockMailChannel_SetLastSeenMessage(ids, message_id) {
+        // deprecated: update mail.channel/fetched_message_id and mail.channel/seen_message_id
         this._mockWrite('mail.channel', [ids, {
             fetched_message_id: message_id,
             seen_message_id: message_id,
         }]);
+        const cps = this._getRecords('mail.channel.partner', [['channel_id', 'in', ids], ['partner_id', '=', this.currentPartnerId]]);
+        this._mockWrite('mail.channel.partner', [cps.map(cp => cp.id), {
+            fetched_message_id: message_id,
+            seen_message_id: message_id,
+        }]);
+    },
+    /**
+     * Simulates `_inverse_channel_partner_ids` on `mail.channel`
+     *
+     * @private
+     */
+     _mockMailChannel__inverseChannelPartnerIds(channelIds) {
+        for (const channelId of channelIds) {
+            const channel = this._getRecords('mail.channel', [['id', 'in', [channelId]]])[0];
+            for (const memberId of channel.channel_partner_ids) {
+                const cp = this._getRecords('mail.channel.partner', [['partner_id', '=', memberId], ['channel_id', '=', channelId]])[0];
+                if (!cp) {
+                    this._mockCreate('mail.channel.partner', {
+                        channel_id: channelId,
+                        partner_id: memberId,
+                    });
+                }
+            }
+        }
```

Comment:
also missing removal of existing member that are gone

Reply:
I don't think we should manage deletion here. Server does it but it's quite obscure how it manages `channel_last_seen_partner_ids`. There is no such handling of this field in the mock server.

If we had some code for `_action_remove_members`, then we would simply remove `mail.channel.partner` record during this action.

Reply:
It's important to do it, it should be declarative (members in the list = actual members), not flux specific.

Just need to find all channel.partner for this channel, and delete those that are not in channel_partner_ids.

Note that for consistency it requires the opposite works too, if that is not yet the case: when creating a channel.partner, it should fill the related channel_partner_ids, and remove from itwhen deleting.